[JIT] Add SchemaCheckMode OpInfo test

ttanpcs opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #82442
- Move test_schema_check to torch/test directory.
- Add opInfo test for SchemaCheckMode to check all operator schemas 
- Add various changes (using isClose instead of equals, skipping complex number cases for certain ops, etc...) in order to have test_schema_check pass.

Differential Revision: [D38437946](https://our.internmc.facebook.com/intern/diff/D38437946)

Commit:
[JIT] Add SchemaCheckMode OpInfo test

[ghstack-poisoned]

Commit:
Update on "[JIT] Add SchemaCheckMode OpInfo test"

- Move test_schema_check to torch/test directory.
- Add opInfo test for SchemaCheckMode to check all operator schemas 
- Add various changes (using isClose instead of equals, skipping complex number cases for certain ops, etc...) in order to have test_schema_check pass.

[ghstack-poisoned]

Commit:
Update on "[JIT] Add SchemaCheckMode OpInfo test"

- Move test_schema_check to torch/test directory.
- Add opInfo test for SchemaCheckMode to check all operator schemas 
- Add various changes (using isClose instead of equals, skipping complex number cases for certain ops, etc...) in order to have test_schema_check pass.

[ghstack-poisoned]

davidberard98 commented on torch/csrc/jit/python/init.cpp:
```diff
@@ -233,10 +233,12 @@ bool loadPythonClasses() {
-  bool is_empty_list =
-      PySequence_Check(self.ptr()) && !PySequence_Size(self.ptr());
-  return is_empty_list;
+c10::optional<IValue> toTypeInferredIValueOptional(py::handle input) {
+  try {
+    return toTypeInferredIValue(input);
+  } catch (const c10::Error& e) {
+    return c10::nullopt;
+  }
```
could you add a comment why we need to catch errors here

davidberard98 commented on torch/csrc/jit/python/init.cpp:
```diff
@@ -1879,16 +1882,21 @@ void initJITBindings(PyObject* module) {
+    c10::optional<IValue> self_value = toTypeInferredIValueOptional(self);
+    c10::optional<IValue> other_value = toTypeInferredIValueOptional(other);
+
+    if (!self_value || !other_value) {
```
maybe also add a comment here (e.g. "only return true if we can be certain that they are aliasing")

davidberard98 reviewed (commented):
looks good, but can you add some comments? lots of special cases here, would be good to document what errors led to these special cases.

Commit:
Update on "[JIT] Add SchemaCheckMode OpInfo test"

- Move test_schema_check to torch/test directory.
- Add opInfo test for SchemaCheckMode to check all operator schemas 
- Add various changes (using isClose instead of equals, skipping complex number cases for certain ops, etc...) in order to have test_schema_check pass.

[ghstack-poisoned]

goldenxuett commented:
@goldenxuett has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38437946).

davidberard98 reviewed (approved):
Looks good!! Remember to land via internal workflow since there are internal changes.

Commit:
Update on "[JIT] Add SchemaCheckMode OpInfo test"

- Move test_schema_check to torch/test directory.
- Add opInfo test for SchemaCheckMode to check all operator schemas 
- Add various changes (using isClose instead of equals, skipping complex number cases for certain ops, etc...) in order to have test_schema_check pass.

Differential Revision: [D38437946](https://our.internmc.facebook.com/intern/diff/D38437946)

[ghstack-poisoned]

Commit:
Update on "[JIT] Add SchemaCheckMode OpInfo test"

- Move test_schema_check to torch/test directory.
- Add opInfo test for SchemaCheckMode to check all operator schemas 
- Add various changes (using isClose instead of equals, skipping complex number cases for certain ops, etc...) in order to have test_schema_check pass.

Differential Revision: [D38437946](https://our.internmc.facebook.com/intern/diff/D38437946)

[ghstack-poisoned]

goldenxuett commented:
@goldenxuett has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38437946).
